expr/func_filterseries.go

```diff
@@ -0,0 +1,84 @@
+package expr
+
+import (
+	"math"
+
+	"github.com/grafana/metrictank/api/models"
+	"github.com/grafana/metrictank/consolidation"
+)
+
+type FuncFilterSeries struct {
+	in        GraphiteFunc
+	fn        string
+	operator  string
+	threshold float64
+}
+
+func NewFilterSeries() GraphiteFunc {
+	return &FuncFilterSeries{}
+}
+
+func (s *FuncFilterSeries) Signature() ([]Arg, []Arg) {
+	return []Arg{
+		ArgSeriesList{val: &s.in},
+		ArgString{key: "func", val: &s.fn, validator: []Validator{IsConsolFunc}},
+		ArgString{key: "operator", val: &s.operator, validator: []Validator{IsOperator}},
+		ArgFloat{key: "threshold", val: &s.threshold},
+	}, []Arg{ArgSeriesList{}}
+}
+
+func (s *FuncFilterSeries) Context(context Context) Context {
+	return context
+}
+
+func (s *FuncFilterSeries) getOperatorFunc() func(float64) bool {
```

Comment:
This could be a free function if it took in s.operator and returned a func that takes 2 floats to compare.

Reply:
Is there a benefit to doing so? Was just trying to keep it clean

Reply:
If another function needs similar functionality, it would be easier to refactor into a utility. I don't think one is necessarily cleaner than the other, but one is definitely more flexible.